src/operationDefinitions/USCoreDocRef/parseParams.ts

```diff
@@ -0,0 +1,239 @@
+                                            },
+                                            display: {
+                                                type: 'string',
+                                            },
+                                        },
+                                        additionalProperties: false,
+                                    },
+                                },
+                                additionalProperties: false,
+                            },
+                        },
+                        additionalProperties: false,
+                    },
+                    {
+                        type: 'object',
+                        required: ['name', 'valueId'],
+                        properties: {
+                            name: {
+                                const: 'patient',
+                            },
+                            valueId: {
+                                type: 'string',
+                            },
+                        },
+                        additionalProperties: false,
+                    },
+                    {
+                        type: 'object',
+                        required: ['name', 'valueDate'],
+                        properties: {
+                            name: {
+                                enum: ['start', 'end'],
+                            },
+                            valueDate: {
+                                type: 'string',
+                            },
+                        },
+                        additionalProperties: false,
+                    },
+                    {
+                        type: 'object',
+                        required: ['name', 'valueBoolean'],
+                        properties: {
+                            name: {
+                                const: 'on-demand',
+                            },
+                            valueBoolean: {
+                                type: 'boolean',
+                            },
+                        },
+                        additionalProperties: false,
+                    },
+                    {
+                        type: 'object',
+                        required: ['name', 'valueId'],
+                        properties: {
+                            name: {
+                                const: 'patient',
+                            },
+                            valueId: {
+                                type: 'string',
+                            },
+                        },
+                        additionalProperties: false,
+                    },
+                ],
+            },
+        },
+    },
+    additionalProperties: {
+        not: true,
+        errorMessage: 'is not a valid property',
+    },
+};
+
+const validatePostParams = ajv.compile(postParamsSchema);
+
+export const parsePostParams = (postParams: any): DocRefParams => {
+    if (validatePostParams(postParams)) {
```

Comment:
Similarly here, we can exit out early by throwing the error.

Reply:
updated